internal/controller/ingressclass_v1beta1_controller.go

```diff
@@ -0,0 +1,248 @@
+	gatewayProxyList := &v1alpha1.GatewayProxyList{}
+	if err := r.List(ctx, gatewayProxyList, client.MatchingFields{
+		indexer.SecretIndexRef: indexer.GenIndexKey(secret.GetNamespace(), secret.GetName()),
+	}); err != nil {
+		r.Log.Error(err, "failed to list gateway proxies by secret", "secret", secret.GetName())
+		return nil
+	}
+
+	// 2. list ingress classes by gateway proxies
+	requests := make([]reconcile.Request, 0)
+	for _, gatewayProxy := range gatewayProxyList.Items {
+		requests = append(requests, r.listIngressClassV1beta1esForGatewayProxy(ctx, &gatewayProxy)...)
+	}
+
+	return distinctRequests(requests)
+}
+
+func (r *IngressClassV1beta1Reconciler) processInfrastructure(tctx *provider.TranslateContext, IngressClassV1beta1 *v1beta1.IngressClass) error {
+	if IngressClassV1beta1.Spec.Parameters == nil {
+		return nil
+	}
+
+	if IngressClassV1beta1.Spec.Parameters.APIGroup == nil ||
+		*IngressClassV1beta1.Spec.Parameters.APIGroup != v1alpha1.GroupVersion.Group ||
+		IngressClassV1beta1.Spec.Parameters.Kind != KindGatewayProxy {
+		return nil
+	}
+
+	// Since v1beta1 does not support specifying the target namespace,
+	// and GatewayProxy is a namespace-scoped resource, we default to using
+	// the "default" namespace for convenience.
+	namespace := "default"
+	if IngressClassV1beta1.Spec.Parameters.Namespace != nil {
+		namespace = *IngressClassV1beta1.Spec.Parameters.Namespace
+	}
+
+	gatewayProxy := new(v1alpha1.GatewayProxy)
+	if err := r.Get(tctx, client.ObjectKey{
+		Namespace: namespace,
+		Name:      IngressClassV1beta1.Spec.Parameters.Name,
+	}, gatewayProxy); err != nil {
+		return fmt.Errorf("failed to get gateway proxy: %w", err)
+	}
+
+	rk := utils.NamespacedNameKind(IngressClassV1beta1)
+
+	tctx.GatewayProxies[rk] = *gatewayProxy
+	tctx.ResourceParentRefs[rk] = append(tctx.ResourceParentRefs[rk], rk)
+
+	// Load secrets if needed
+	if gatewayProxy.Spec.Provider != nil && gatewayProxy.Spec.Provider.ControlPlane != nil {
+		auth := gatewayProxy.Spec.Provider.ControlPlane.Auth
+		if auth.Type == v1alpha1.AuthTypeAdminKey && auth.AdminKey != nil && auth.AdminKey.ValueFrom != nil {
+			if auth.AdminKey.ValueFrom.SecretKeyRef != nil {
+				secretRef := auth.AdminKey.ValueFrom.SecretKeyRef
+				secret := &corev1.Secret{}
+				if err := r.Get(tctx, client.ObjectKey{
+					Namespace: namespace,
+					Name:      secretRef.Name,
+				}, secret); err != nil {
+					r.Log.Error(err, "failed to get secret for gateway proxy", "namespace", namespace, "name", secretRef.Name)
+					return err
+				}
+				tctx.Secrets[client.ObjectKey{
+					Namespace: namespace,
+					Name:      secretRef.Name,
+				}] = secret
+			}
+		}
+	}
+
+	if service := gatewayProxy.Spec.Provider.ControlPlane.Service; service != nil {
+		if err := addProviderEndpointsToTranslateContext(tctx, r.Client, types.NamespacedName{
+			Namespace: gatewayProxy.GetNamespace(),
+			Name:      service.Name,
+		}); err != nil {
+			return err
+		}
+	}
```

Comment:
It is necessary to determine whether the call here is accurate. In subsequent functions, resolveServiceEndpoints(tctx, c, serviceNN, true, nil)

the param `supportsEndpointSlice` We set it to true by default, which theoretically would cause problems on k8s clusters that do not support `EndpointSlice`